### PR TITLE
[NIC-29] Add get logic to watcher-service

### PR DESCRIPTION
**Pull request by AksyonovaKaterina**

### CODE REVIEW

File: backend/watcher-service/src/main/resources/http/watcherControllerRequests.http [hunk 1]
```diff
@@ -1,6 +1,7 @@
...
-Authorization: Bearer eyJraWQiOiIxNDQwNzcyMTY5NDQyNDUwMjM2NCIsImFsZyI6IlJTMjU2In0.eyJpc3MiOiJub3NxbC1pdC1jb21wYW55Iiwic3ViIjoiNjM4NTFmYTViN2YzZGEwOTExYjgxZWQzIiwiaWF0IjoxNjY5NjY4Nzg3LCJleHAiOjE2Njk2NzA1ODcsInJvbGUiOiJERVZFTE9QRVIifQ.fu5DX_UaeiQjGdeyenwmIz9s3PIS8hvK3pqMVnJee0CkdI4Cg9NwqhgAKKHuzlApHEXlxlheqrQNs1plY-KYbBPyXUOcCb0-TF3CjVnmRY8j_kFg5uH2SR70YoWAr-xU_3a_9ezRiFUzJgKRE2cZ6cwJ4cfqj9kOhm8nhrQBTjDVxipDg3FQLqs12rPVu5anUmhSIFoF0z4bWUAVLC_GmVG8SuRziPCYLj4cstFdMP93EF7vjzufqJplkfYE-b8tc0yjDcManymlwIcAkFAWwwqGnMGejgpD5_BhV3v5sn28nDQ1DFeegllhiIgyQPs2p2zkHDVdR-2xLmW_8Ui66Q
+#Authorization: Bearer <access_token>
+Authorization: Bearer eyJraWQiOiIxNDQwNzcyMTY5NDQyNDUwMjM2NCIsImFsZyI6IlJTMjU2In0.eyJpc3MiOiJub3NxbC1pdC1jb21wYW55Iiwic3ViIjoiNjM4NTFmYTViN2YzZGEwOTExYjgxZWQ1IiwiaWF0IjoxNjY5NjczNzExLCJleHAiOjE2ODUyMjU3MTEsInJvbGUiOiJIUiJ9.JdDF9BfqZoEFnplSEXKlgJRjYxGmW5EqLjAAay7Z3N1hE0eJ1nfeAs3n9ERcjRX2ScbUkY4bqpfawP8AZbg6lmZSn1_WYtwmLesVhPlesJdo7U9plBwWIJth1GLMa4qJpOzNHaVIST7un7sT_5q2shnb9K1BQacwv4p0SrE2mpM_PDiePN-4ZYYG7RXNC2lbAJsTyY7lMTH4nl620ve8qComDL4bbaSsr7lnUPz-4SmSYKHSqEP952WKlbxrBA2WlnJE3lSgXYIxu5NuLmu7TSexOjD4Y-ykP0d3Y8r1xBZk_2BZijpR2I1_Mjfg7fTdtxVxe0eZlexebZXdZ7J5Mw
```
Comment (V7074) [inline backend/watcher-service/src/main/resources/http/watcherControllerRequests.http]: token?

File: backend/watcher-service/src/main/resources/http/watcherControllerRequests.http [hunk 2]
```diff
@@ -9,4 +10,22 @@ Content-Type: application/json
...
+Content-Type: application/json
+
+### Get all watcher
+GET http://localhost:8082/public/api/v1/watcher/all?page=0
+#Authorization: Bearer <access_token>
+Authorization: Bearer eyJraWQiOiIxNDQwNzcyMTY5NDQyNDUwMjM2NCIsImFsZyI6IlJTMjU2In0.eyJpc3MiOiJub3NxbC1pdC1jb21wYW55Iiwic3ViIjoiNjM4NTFmYTViN2YzZGEwOTExYjgxZWQ1IiwiaWF0IjoxNjY5NjczNzExLCJleHAiOjE2ODUyMjU3MTEsInJvbGUiOiJIUiJ9.JdDF9BfqZoEFnplSEXKlgJRjYxGmW5EqLjAAay7Z3N1hE0eJ1nfeAs3n9ERcjRX2ScbUkY4bqpfawP8AZbg6lmZSn1_WYtwmLesVhPlesJdo7U9plBwWIJth1GLMa4qJpOzNHaVIST7un7sT_5q2shnb9K1BQacwv4p0SrE2mpM_PDiePN-4ZYYG7RXNC2lbAJsTyY7lMTH4nl620ve8qComDL4bbaSsr7lnUPz-4SmSYKHSqEP952WKlbxrBA2WlnJE3lSgXYIxu5NuLmu7TSexOjD4Y-ykP0d3Y8r1xBZk_2BZijpR2I1_Mjfg7fTdtxVxe0eZlexebZXdZ7J5Mw
```
Comment (V7074) [inline backend/watcher-service/src/main/resources/http/watcherControllerRequests.http]: token?

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt [hunk 3]
```diff
@@ -2,22 +2,59 @@ package com.nosql.watcher.controller
...
 @RestController
 @RequestMapping(PUBLIC_API_V1_WATCHER_URL_PATH)
+@SecurityRequirement(name = SECURITY_SCHEME_IDENTIFIER)
 class WatcherController(
     private val watcherService: WatcherService,
 ) {
 
+    @GetMapping(
+        produces = [APPLICATION_JSON_VALUE],
+    )
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt]: maybe use `/all` suffix path?

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/watcher-service/src/main/resources/http/watcherControllerRequests.http (hunk 1 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/resources/http/watcherControllerRequests.http]: fixed

File: backend/watcher-service/src/main/resources/http/watcherControllerRequests.http (hunk 2 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/resources/http/watcherControllerRequests.http]: fixed

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt (hunk 3 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt]: fixed

**Review by V7074 — APPROVED** [review]

**Pull request merged**